core/src/odbc_uri.rs

```diff
@@ -947,20 +1110,41 @@ mod unit {
             );
         }
 
+        #[tokio::test(flavor = "current_thread")]
+        async fn ldap_correctness() {
+            use crate::odbc_uri::ODBCUri;
+            let odbc_str = "URI=mongodb://localhost/abc?authSource=$external&authMechanism=PLAIN;UID=foo;PWD=bar";
+            let actual = ODBCUri::new(odbc_str.to_string())
+                .unwrap()
+                .try_into_client_options()
+                .await
+                .unwrap()
+                .client_options
+                .credential
+                .unwrap();
+            assert_eq!(
+                Some(mongodb::options::AuthMechanism::Plain),
+                actual.mechanism
+            );
+            assert_eq!(Some("$external".to_string()), actual.source);
+            assert_eq!(Some("foo".to_string()), actual.username);
+        }
+
         #[tokio::test(flavor = "current_thread")]
         async fn auth_source_correctness() {
             use crate::odbc_uri::ODBCUri;
-            for (source, uri) in [
+            for (expected, uri) in [
                 (Some("authDB".to_string()), "URI=mongodb://localhost/?authSource=authDB;UID=foo;PWD=bar"),
-                (None, "URI=mongodb://localhost/;UID=foo;PWD=bar"),
+                (Some("admin".to_string()), "URI=mongodb://localhost/;UID=foo;PWD=bar"),
```

Comment:
Ended up being more correct, since the default authSource is admin if it's not specified elsewhere!